### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Khaled-AlKhateeb**
##Tasks Done ✔️ 

- Refactored the Calculator component from a class based component to a functional component with hooks.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,59 +1,49 @@
...
 
-  getValue = (e) => {
-    const value = e.target.innerHTML;
-    const obj = calculate(this.state, value);
-    this.setState(obj);
-  }
+  useState = () => {
+    const obj = calculate(count, setCount);
+    setCount(obj);
+  };
```
Comment (prolajumokeoni) [inline src/components/Calculator.js]: - [x] Please ensure that your app still works properly after the refactoring from class to functional-based components, currently when your app starts it shows a blank page, please fix it 👍🏿 

Screenshot_512

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Khaled-AlKhateeb 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Khaled-AlKhateeb, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Khaled-AlKhateeb** [comment]
Thank you!

**Pull request merged**